src/Spryker/Zed/AppPayment/Business/Payment/Message/MessageSender.php

```diff
@@ -32,22 +32,20 @@
 use Spryker\Shared\Kernel\Transfer\TransferInterface;
 use Spryker\Zed\AppKernel\AppKernelConfig;
 use Spryker\Zed\AppPayment\AppPaymentConfig;
-use Spryker\Zed\AppPayment\Dependency\Facade\AppPaymentToAppKernelFacadeInterface;
 use Spryker\Zed\AppPayment\Dependency\Facade\AppPaymentToMessageBrokerFacadeInterface;
 
 class MessageSender
 {
     public function __construct(
         protected AppPaymentToMessageBrokerFacadeInterface $appPaymentToMessageBrokerFacade,
-        protected AppPaymentConfig $appPaymentConfig,
-        protected AppPaymentToAppKernelFacadeInterface $appPaymentToAppKernelFacade
+        protected AppPaymentConfig $appPaymentConfig
     ) {
     }
 
     public function sendAddPaymentMethodMessage(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
     {
-        // Only send the message when the AppConfig is in state NEW.
-        if ($appConfigTransfer->getStatus() === AppKernelConfig::APP_STATUS_CONNECTED) {
+        // Do not send the message when App is in state "disconnected" or when the app is marked as inactive.
```

Comment:
is it OK to send the message `AddPaymentMethod` every time when the app config is saved/re-saved?

NOTE: on SCOS side it will be handled [correctly](https://github.com/spryker/payment/blob/master/src/Spryker/Zed/Payment/Business/Method/PaymentMethodUpdater.php#L173-L198)

Reply:
I think, yes.